examples/api/test/widgets/implicit_animations/animated_fractionally_sized_box.0_test.dart

```diff
@@ -0,0 +1,125 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_api_samples/widgets/implicit_animations/animated_fractionally_sized_box.0.dart'
+    as example;
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  testWidgets(
+    'AnimatedFractionallySizedBox animates on tap',
+    (WidgetTester tester) async {
+      await tester.pumpWidget(
+        const example.AnimatedFractionallySizedBoxExampleApp(),
+      );
+
+      final Finder fractionallySizedBoxFinder = find.descendant(
+        of: find.byType(AnimatedFractionallySizedBox),
+        matching: find.byType(FractionallySizedBox),
+      );
+
+      const double beginWidthFactor = 0.75;
+      const double endWidthFactor = 0.25;
+      const double beginHeightFactor = 0.25;
+      const double endHeightFactor = 0.75;
+      const Alignment beginAlignment = Alignment.bottomRight;
+      const Alignment endAlignment = Alignment.topLeft;
+
+      FractionallySizedBox fractionallySizedBox = tester.widget(
+        fractionallySizedBoxFinder,
+      );
+      expect(fractionallySizedBox.widthFactor, beginWidthFactor);
+      expect(fractionallySizedBox.heightFactor, beginHeightFactor);
+      expect(fractionallySizedBox.alignment, beginAlignment);
+
+      // Tap on the AnimatedFractionallySizedBoxExample to start the forward
+      // animation.
+      await tester.tap(
+        find.byType(example.AnimatedFractionallySizedBoxExample),
+      );
+      await tester.pump();
+
+      fractionallySizedBox = tester.widget(fractionallySizedBoxFinder);
+      expect(fractionallySizedBox.widthFactor, beginWidthFactor);
+      expect(fractionallySizedBox.heightFactor, beginHeightFactor);
+      expect(fractionallySizedBox.alignment, beginAlignment);
+
+      // Advance animation to the middle.
+      await tester.pump(
+        example.AnimatedFractionallySizedBoxExampleApp.duration ~/ 2,
+      );
+
+      final double t =
```

Comment:
Nit: Mayyybe give this a more explicit name. It took me awhile to track down what `t` was when I saw it used below, though admittedly that's only here in the GitHub UI that it's a problem since I would have used jump to definition in my editor. Maybe `curveHalf` or something?

It could be argued that this styleguide rule applies or does not apply here: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-abbreviations

Reply:
> Nit: Mayyybe give this a more explicit name. It took me awhile to track down what `t` was when I saw it used below, though admittedly that's only here in the GitHub UI that it's a problem since I would have used jump to definition in my editor. Maybe `curveHalf` or something?
> 
> It could be argued that this styleguide rule applies or does not apply here: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-abbreviations

t is a variable typically used to represent the animation value in calls to `lerp`. Since the animation in this case isn't using any particularly confusing arithmetic, I think it would fine to just leave it as is.